src/core/lib/draw-layers.js

```diff
@@ -186,8 +186,13 @@ function drawLayersInViewport(gl, {
         {layerIndex}
       );
 
+      // All parameter resolving is done here instead of the layer
       // Blend parameters must not be overriden
-      const layerParameters = Object.assign({viewport: glViewport}, layer.props.parameters || {});
+      const layerParameters = Object.assign({}, layer.props.parameters || {}, parameters);
+
+      Object.assign(layerParameters, {
```

Comment:
Why not merge with the line above?

Reply:
> Why not merge with the line above?

Readability, I wanted to show more clearly the sequence of things. I was going back and forth, both ways are OK.

Since you are about to do another pass I'll merge and let you change it if you want to.